src/transformers/data/data_collator.py

```diff
@@ -1611,3 +1611,30 @@ def numpy_mask_tokens(self, inputs: Any) -> Tuple[Any, Any, Any, Any]:
             ) & masked_indices[i]
 
         return inputs.astype(np.int64), perm_mask, target_mapping, labels.astype(np.int64)
+
+@dataclass
+class DataCollatorWithFlattening(DefaultDataCollator):
+    return_tensors: str = "pt"
+    def __init__(self, *args, return_position_ids=True, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.return_position_ids=return_position_ids
+        warnings.warn(
+            "Using `DataCollatorWithFlattening` will flatten the entire mini batch into single long sequence."
+            "Make sure your attention computation is able to handle it!"
+        )
```

Comment:
How should a user make sure?

Reply:
no easy way, the user will have to make sure the model supports FA2 and have `position_ids`
until the model accepting `cu_seq_len` which will allow every model remove padding once for all, i think this serve as a good warning